### ISSUE TITLE
'sevctl ok' and kernel reporting different status for BIOS SEV setup

### ISSUE THREAD

**Issue by cclaudio** [virtee/sevctl#130]
I was using the `sevctl ok` command to check if SEV is enabled in the BIOS and the output below suggests that it is; however, a dmesg message says otherwise ` ccp 0000:24:00.1: SEV: memory encryption not enabled by BIOS`.  I am wondering if the kernel ccp and sevctl are using different methods to check if the SEV is enabled in the BIOS.

I think the SEV-SNP check may also have issues because it is returning PASS, but this AMD milan system has hardware support only for SEV and SEV-ES, and AFAIK they were not enabled in the BIOS.

The system is running RHEL 8.7 (kernel 4.18.0) and sevctl-0.3.0. 

```
[cclaudio@milan ~]$ sudo sevctl ok
[ PASS ] - AMD CPU
[ PASS ]   - Microcode support
[ PASS ]   - Secure Memory Encryption (SME)
[ PASS ]   - Secure Encrypted Virtualization (SEV)
[ PASS ]     - Encrypted State (SEV-ES)
[ PASS ]     - Secure Nested Paging (SEV-SNP)
[ PASS ]       - VM Permission Levels
[ PASS ]         - Number of VMPLs: 4
[ PASS ]     - Physical address bit reduction: 5
[ PASS ]     - C-bit location: 51
[ PASS ]     - Number of encrypted guests supported simultaneously: 509
[ PASS ]     - Minimum ASID value for SEV-enabled, SEV-ES disabled guest: 1
[ FAIL ]     - SEV enabled in KVM: Error - contents read from /sys/module/kvm_amd/parameters/sev: N
[ FAIL ]     - Reading /dev/sev: /dev/sev not readable: No such file or directory (os error 2)
[ FAIL ]     - Writing /dev/sev: /dev/sev not writable: No such file or directory (os error 2)
[ PASS ]   - Page flush MSR: ENABLED
[ PASS ] - KVM supported: API version: 12
[ PASS ] - Memlock resource limit: Soft: 65536 | Hard: 65536
error: One or more tests in sevctl-ok reported a failure
[cclaudio@milan ~]$
```

**Comment by larrydewey** [comment]
@cclaudio I have a [branch](https://github.com/larrydewey/sevctl/tree/syscfg-msr) with a patch proposal. Would you mind giving it a run to see if you see `FAIL` now?

**Comment by cclaudio** [comment]
@larrydewey I tested your branch in another milan system. It has hardware support for SEV, SEV-ES and SEV-SNP, and they are all enabled in the BIOS.

With your branch, `sevctl ok` is failing and skipping some test cases, see below. With the main branch, no tests is skipped and they all pass.

```
cclaudio@milan:~/src/larry/sevctl$ sudo ./target/debug/sevctl ok
[ PASS ] - AMD CPU
[ PASS ]   - Microcode support
[ FAIL ]   - Secure Memory Encryption (SME)
[ PASS ]   - Secure Encrypted Virtualization (SEV)
[ PASS ]     - Encrypted State (SEV-ES)
[ FAIL ]     - Secure Nested Paging (SEV-SNP)
[ SKIP ]       - VM Permission Levels
[ SKIP ]         - Number of VMPLs
[ PASS ]     - Physical address bit reduction: 5
[ PASS ]     - C-bit location: 51
[ PASS ]     - Number of encrypted guests supported simultaneously: 509
[ PASS ]     - Minimum ASID value for SEV-enabled, SEV-ES disabled guest: 100
[ PASS ]     - SEV enabled in KVM: enabled
[ PASS ]     - SEV-ES enabled in KVM: enabled
[ PASS ]     - Reading /dev/sev: /dev/sev readable
[ PASS ]     - Writing /dev/sev: /dev/sev writable
[ PASS ]   - Page flush MSR: ENABLED
[ PASS ] - KVM supported: API version: 12
[ PASS ] - Memlock resource limit: Soft: 67108864 | Hard: 67108864
Error: One or more tests in sevctl-ok reported a failure
```

**Comment by larrydewey** [comment]
Added a change in the [msru pr](https://github.com/virtee/msru/pull/3) which will return a specific error when the device file is not present.

**Comment by larrydewey** [comment]
PR merged, and msru v0.2.0 released. Just waiting on the next release of sevctl

**Comment by tylerfanelli** [comment]
@larrydewey @cclaudio Releasing a new sevctl version now. Has this issue been fixed?
